llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -394,7 +394,6 @@ def is_done(self) -> bool:
     async def _done(self, ctx: Context, ev: StopEvent) -> None:
         """Tears down the whole workflow and stop execution."""
         ctx._retval = ev.result or None
-        ctx.write_event_to_stream(ev)
```

Comment:
Removing this, since we're actually writing `StopEvent` twice to the stream.

The other time a `StopEvent` gets written to the stream is under the execution of `_run_workflow()` within `Workflow.run()`:

https://github.com/run-llama/llama_index/blob/a18b94699ac4e49b17f3f49879adf29dfc7c3ed3/llama-index-core/llama_index/core/workflow/workflow.py#L331

Reply:
I'm actually pretty confused by this logic -- the user is the one emiting the stop event, so the one the user emits is probably the one we want to write to the stream (i.e. the one in _done right?)

Reply:
Yeah, I think that's fair. I just arbitrarily chose where to remove a `StopEvent` emission, but I think your logic makes sense. Based on that, I'd adjust this PR by keeping that one in favour of removing this one:

https://github.com/run-llama/llama_index/blob/a18b94699ac4e49b17f3f49879adf29dfc7c3ed3/llama-index-core/llama_index/core/workflow/workflow.py#L331

Reply:
This was done to avoid the need for the user to send a stop event to the stream. Using the stop event to signal "ok we're done streaming" is an implementation detail that should be kept hidden if possible

Reply:
(to say as long as we send it transparently, I'm fine removing any of the two, pick the one that looks less intuitive)

Reply:
Thanks @masci and @logan-markewich. Based on this discussion, I've updated the PR to remove the emission of `StopEvent` under `_run_workflow()`.

Reply:
Looks good to me :)